library/classes/Document.class.php

```diff
@@ -587,6 +631,14 @@ function createDocument($patient_id, $category_id, $filename, $mimetype, &$data,
       if (file_put_contents($filepath . $filename, $data) === FALSE) {
         return xl('Failed to create') . " $filepath$filename";
       }
+	  if( $has_thumbnail ) {
+		 $this->thumb_url = "file://" . $filepath . $this->get_thumb_name($filename);
+		 // Store the file into its proper directory.
+		 if (file_put_contents($filepath . 'th_' . $filename, $thumbnail_data) === FALSE) {
```

Comment:
Note I did not address this issue in the code I brought into the codebase. Please submit a separate commit to fix this issue.